libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/MatrixArea/MatrixArea.tsx

```diff
@@ -428,7 +484,24 @@ export class MatrixArea extends React.PureComponent<
     }
     const index = j + i * rowLength;
     selectedCells[index] = !selectedCells[index];
-    this.setState({ selectedCells });
+    let disableClearAll = false;
+    let disableSelectAll = false;
+    if (selectedCells.every((value) => value)) {
+      disableClearAll = false;
+      disableSelectAll = true;
+    } else if (selectedCells.every((value) => !value)) {
+      disableClearAll = true;
+      disableSelectAll = false;
+    }
+    this.setState({
+      disableClearAll,
+      disableSelectAll,
+      selectedCells
+    });
+    this.updateStateFromSelectedCells(selectedCells);
+  }
+
```

Comment:
this file seems too big, are we able to split it?
We have lint rule for 300

Reply:
done!